setup.py

```diff
@@ -47,6 +47,8 @@ class ExtraDependencies:
     ]
     grid = [
         "gridx-prescient>=2.2.1",  # idaes.tests.prescient
+        "scikit-learn>=1.1",
```

Comment:
We need to have some discussion of this at least before we go ahead. Adding dependencies makes maintenance harder and adds more potential points of conflict. Could you provide some discussion here of why these are needed and cannot be provided by existing dependencies (@lbianchi-lbl will want to hear this too)?

One possible solution we should discuss is whether this could and should live in a separate repository where it has greater control over its environment (with out impacting the rest of IDAES) and also means that ongoing maintenance of this code has less effect on IDAES overall.

Reply:
Scikit-learn (sklearn) is needed to import Kmeans, which is a [clustering technique](https://scikit-learn.org/1.5/modules/clustering.html) used to determine the sum of squares error for a given number of clusters in a given dataset (price signals in this case). Kneed is needed to import KneeLocator and uses the output of Kmeans to determine what the optimal number of clusters is that a dataset should be broken down into

As far as I know, no existing dependencies cover clustering techniques

Reply:
@andrewlee94 @lbianchi-lbl - here is a reminder of the old comment thread related to this: https://github.com/IDAES/idaes-pse/pull/1358/files#r1653251726

Reply:
From a quick glance, `scipy` appears to have some clustering capability (https://docs.scipy.org/doc/scipy/reference/cluster.html).

My main concern here is that we have had issues in the past with dependency conflicts creating issues with testing the tools (in that a failure in one optional dependency, even if only used in one small part of the code, can break the testing for the entire codebase).

Reply:
I'll look into this...

Reply:
I've implemented the tools from scipy